driver-core/src/main/com/mongodb/binding/ClusterBinding.java

```diff
@@ -67,7 +83,7 @@ public ConnectionSource getReadConnectionSource() {
 
     @Override
     public SessionContext getSessionContext() {
-        return NoOpSessionContext.INSTANCE;
+        return new NoOpSessionContext(readConcern);
```

Comment:
This is a bit of a hack.  Could instead add a subclass of NoOpSessionContext that manages the read preference.

Reply:
oops... Meant read concern